x-pack/test_serverless/api_integration/test_suites/observability/ai_assistant/tests/complete/complete.spec.ts

```diff
@@ -0,0 +1,524 @@
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const log = getService('log');
+  const svlUserManager = getService('svlUserManager');
+  const svlCommonApi = getService('svlCommonApi');
+
+  const observabilityAIAssistantAPIClient = getService('observabilityAIAssistantAPIClient');
+
+  const COMPLETE_API_URL = `/internal/observability_ai_assistant/chat/complete`;
+
+  const messages: Message[] = [
+    {
+      '@timestamp': new Date().toISOString(),
+      message: {
+        role: MessageRole.System,
+        content: 'You are a helpful assistant',
+      },
+    },
+    {
+      '@timestamp': new Date().toISOString(),
+      message: {
+        role: MessageRole.User,
+        content: 'Good morning, bot!',
+        // make sure it doesn't 400 on `data` being set
+        data: '{}',
+      },
+    },
+  ];
+
+  describe('/internal/observability_ai_assistant/chat/complete', () => {
+    let proxy: LlmProxy;
+    let connectorId: string;
+    let roleAuthc: RoleCredentials;
+    let internalReqHeader: InternalRequestHeader;
+
+    async function getEvents(
+      params: { screenContexts?: ObservabilityAIAssistantScreenContextRequest[] },
+      cb: (conversationSimulator: LlmResponseSimulator) => Promise<void>
+    ) {
+      const titleInterceptor = proxy.intercept('title', (body) => isFunctionTitleRequest(body));
+
+      const conversationInterceptor = proxy.intercept(
+        'conversation',
+        (body) => !isFunctionTitleRequest(body)
+      );
+      const responsePromise = new Promise<Response>((resolve, reject) => {
+        supertestWithoutAuth
+          .post(COMPLETE_API_URL)
+          .set(roleAuthc.apiKeyHeader)
+          .set(internalReqHeader)
+          .send({
+            messages,
+            connectorId,
+            persist: true,
+            screenContexts: params.screenContexts || [],
+          })
+          .then((response) => resolve(response))
+          .catch((err) => reject(err));
+      });
+
+      const [conversationSimulator, titleSimulator] = await Promise.all([
+        conversationInterceptor.waitForIntercept(),
+        titleInterceptor.waitForIntercept(),
+      ]);
+
+      await titleSimulator.status(200);
+      await titleSimulator.next('My generated title');
+      await titleSimulator.complete();
+
+      await conversationSimulator.status(200);
+      await cb(conversationSimulator);
+
+      const response = await responsePromise;
+
+      return (
+        String(response.body)
+          .split('\n')
+          .map((line) => line.trim())
+          .filter(Boolean)
+          .map((line) => JSON.parse(line) as StreamingChatResponseEvent)
+          // Filter BufferFlush events that appear if isCloudEnabled is true which is the case in serverless tests
+          .filter((event) => event.type !== StreamingChatResponseEventType.BufferFlush)
```

Comment:
added filtering out bufferFlush events so as not to break [test below](https://github.com/elastic/kibana/pull/192219/files#diff-bf995cd531c095c8a98833ec341c5ce3971c0d1467220974093d7127beb25418R288) that checks for the 4th even which gets pushed down in serverless from the BufferFlush event.